corehq/apps/couch_sql_migration/couchsqlmigration.py

```diff
@@ -617,3 +654,71 @@ def commit_migration(domain_name):
         assert should_use_sql_backend(domain_name)
     datadog_counter("commcare.couch_sql_migration.total_committed")
     _logger.info("committed migration for {}".format(domain))
+
+
+class PartiallyLockingQueue(object):
+    def __init__(self, key_base):
+        self.queue_by_id = defaultdict(deque)
+        self.key_base = "%s.%%s.%%s" % key_base  # <key_base>.<lock_type>.<obj_id>
+        self.redis = get_redis_client()
```

Comment:
since this is only running on a single machine can we not just use local locking?